lib/chef/provider/user.rb

```diff
@@ -108,7 +108,7 @@ def define_resource_requirements
             "Note that user update converge may report false-positive on the basis of mismatched password. "
         end
         requirements.assert(:all_actions) do |a|
-          a.assertion { !((new_resource.expire_date || new_resource.inactive) && platform_family?(%w{aix freebsd mac_os_x netbsd openbsd openindiana solaris2 windows})) }
+          a.assertion { !((new_resource.expire_date || new_resource.inactive) && linux?) }
```

Comment:
That's the opposite of what you want/ You want
```ruby
          a.assertion { !((new_resource.expire_date || new_resource.inactive) && !linux?) }
```

Which is a weird way of writing:

```ruby
          # either neither linux-only value is set, or we need to be on Linux.
          a.assertion { (!new_resource.expire_date && !new_resource.inactive) || !linux?) }
```

Reply:
Hmmmm - wonder if we can wrap a test around this...

Reply:
good catch, thanks. change made.